modules/actor/actor-wfrp4e.js

```diff
@@ -1283,125 +1283,155 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
 
-    injury = duplicate(injury)
-    injury.system.duration.value--
+    injury = foundry.utils.duplicate(injury);
+    injury.system.duration.value--;
 
-    if (injury.system.duration.value < 0)
+    if (injury.system.duration.value <= 0) {
       injury.system.duration.value = 0;
-
-    if (injury.system.duration.value == 0) {
-      let chatData = game.wfrp4e.utility.chatDataSetup(game.i18n.format("CHAT.InjuryFinish", { injury: injury.name }), "gmroll")
-      chatData.speaker = { alias: this.name }
-      ChatMessage.create(chatData)
+      let chatData = game.wfrp4e.utility.chatDataSetup(game.i18n.format("CHAT.InjuryFinish", { injury: injury.name }), "gmroll");
+      chatData.speaker = { alias: this.name };
+      ChatMessage.create(chatData);
     }
-    this.updateEmbeddedDocuments("Item", [injury]);
-  }
 
+    await this.updateEmbeddedDocuments("Item", [injury]);
+  }
 
+  /**
+   * Decrements all diseases of this Actor
+   *
+   * @return {Promise<void>}
+   */
   async decrementDiseases() {
-    this.diseases.forEach(d => this.decrementDisease(d))
+    for (let d of this.diseases)
+      await this.decrementDisease(d);
   }
 
+  /**
+   * Decrements a value for Disease's `incubation` or `duration` attribute, depending on whether it is active or not.
+   * If value reaches 0, activates (if `incubation`), or finishes (if `duration`) the Disease.
+   *
+   * @param {ItemWfrp4e} disease
+   *
+   * @return {Promise<void>}
+   */
   async decrementDisease(disease) {
-    let d = duplicate(disease)
-    if (!d.system.duration.active) {
-      if (Number.isNumeric(d.system.incubation.value)) {
-
-        d.system.incubation.value--
-        if (d.system.incubation.value <= 0) {
-          this.activateDisease(d)
-          d.system.incubation.value = 0;
-        }
-      }
-      else {
-        let chatData = game.wfrp4e.utility.chatDataSetup(`Attempted to decrement ${d.name} incubation but value is non-numeric`, "gmroll", false)
-        chatData.speaker = { alias: this.name }
-        ChatMessage.create(chatData)
-      }
-    }
-    else {
-      if (Number.isNumeric(d.system.duration.value)) {
+    let d = foundry.utils.duplicate(disease);
+    let type = d.system.duration.active ? 'duration' : 'incubation';
 
-        d.system.duration.value--
-        if (d.system.duration.value == 0)
-          this.finishDisease(d)
-      }
-      else {
-        let chatData = game.wfrp4e.utility.chatDataSetup(`Attempted to decrement ${d.name} duration but value is non-numeric`, "gmroll", false)
-        chatData.speaker = { alias: this.name }
-        ChatMessage.create(chatData)
+    if (Number.isNumeric(d.system[type].value)) {
+      d.system[type].value--;
+
+      if (d.system[type].value <= 0) {
+        d.system[type].value = 0;
+
+        if (type === 'incubation')
+          await actor.activateDisease(d);
+
+        if (type === 'duration')
+          await actor.finishDisease(d);
```

Comment:
Added `await`, since we are updating the disease further in the same function, so we don't lose the data.